src/bio/bio_xstream.c

```diff
@@ -1029,23 +942,18 @@ bio_xsctxt_alloc(struct bio_xs_context **pctxt, int xs_id)
 	 * spdk_bdev_initialize() call, it also could be used for blobstore
 	 * metadata io channel in following init_bio_bdevs() call.
 	 */
-	ctxt->bxc_msg_ring = spdk_ring_create(SPDK_RING_TYPE_MP_SC,
-					      DAOS_MSG_RING_SZ,
-					      SPDK_ENV_SOCKET_ID_ANY);
-	if (ctxt->bxc_msg_ring == NULL) {
-		D_ERROR("failed to allocate msg ring\n");
-		rc = -DER_NOMEM;
-		goto out;
-	}
-
-	snprintf(name, sizeof(name), "daos_spdk_%d", xs_id);
-	ctxt->bxc_thread = spdk_allocate_thread(send_msg, start_poller,
-						stop_poller, ctxt, name);
+	snprintf(th_name, sizeof(th_name), "daos_spdk_%d", xs_id);
+	ctxt->bxc_thread = spdk_thread_create((const char *)th_name, NULL);
 	if (ctxt->bxc_thread == NULL) {
 		D_ERROR("failed to alloc SPDK thread\n");
 		rc = -DER_NOMEM;
+		if (nvme_glb.bd_init_thread == NULL) {
+			spdk_thread_lib_fini();
+			spdk_env_fini();
+		}
 		goto out;
 	}
+	spdk_set_thread(ctxt->bxc_thread);
```

Comment:
is this something that the original spdk_alloc_thread() did by default before?

Reply:
Yes, exactly.